board/safety/safety_subaru.h

```diff
@@ -165,7 +187,27 @@ static int subaru_tx_hook(CANPacket_t *to_send) {
     violation |= steer_torque_cmd_checks(desired_torque, -1, limits);
   }
 
-  if (violation){
+  if (subaru_longitudinal) {
```

Comment:
this shouldn't be needed since this only further blacklists messages. if we're not doing long, tx should already be false from the `msg_allowed` check

Reply:
done

Reply:
I think we do want this check actually, because we use ES_Distance for a cancel request even when stock long is enabled, and it might get blocked here if eyesight happens to be doing the max accel (I've seen it go to 4000)

Reply:
that shouldn't be allowed in stock long. can you make a pair of PRs to openpilot and panda to fix that? no actuation should be allowed while cancelling

Reply:
Done